README.md

```diff
@@ -11,4 +11,73 @@ However, if exceptions occur frequently, consider adjusting guidelines (by creat
 
 ## Table of contents
 
-// TODO
+- [Git](#git)
+  - [Workflow](#workflow)
+  - [Commit messages](#commit-messages)
+
+## [Git](#git)
+
+### [Workflow](#workflow)
+
+1. Create a feature branch  
+
+    - Prefix branch name with `feature/` or `fix/` when applicable; otherwise, prefix with your initials, e.g. `tk/`
+    - Use kebabcase
+    - Include the issue number when applicable
+    - Briefly describe the change  
+
+    Examples: `feature/610-seelog-wrapper`, `fix/tequilapi-startup`, `tk/polish`
+
+2. Implement the changes (series of commits)
+    - It's best when commmits are clearly separated, i.e. each fix, change, refactoring, extra tests go into their own commits
+    - When fixing up a commit that already exists on your feature branch, try your best to squash the fix to the related commit so that we can avoid noise in git history
+      - If it's the last commit:
+
+        ```
+        $ git commit --amend
+        ```
+
+      - Otherwise, do [an interactive rebase](https://help.github.com/en/articles/about-git-rebase) to squash them into cohesive commits with good messages:
+
+        ```
+        $ git fetch origin
+        $ git rebase -i origin/master
+        ```
+3. Rebase frequently to incorporate upstream changes
+
+    ```
+    $ git pull --rebase origin master
+    ```
+
+4. Push your branch and create a pull request (PR)
+    - Request reviews as suggested and anyone else you see fit
+    - Wait for approvals and CI checks to pass
+
+5. Merge PR and remove the branch (or ask someone with push rights to do it)
+
+General tips:
+
+- Try to integrate with master branch ASAP. Prefer delivering feature in smaller working chunks rather than one huge PR.
+- Avoid merging `origin/master` into feature branch. Rebase instead to keep git history as linear as possible.
+
+### [Commit messages](#commit-messages)
+
+We use widely established conventions for idiomatic Git commit messages:
+
+- Use imperative mood in the subject line.  
```

Comment:
We can add an explanation about why one should use imperative mood - i.e. this has quite a good reference to git itself: https://stackoverflow.com/questions/3580013/should-i-use-past-or-present-tense-in-git-commit-messages/3580764#3580764

Reply:
There's a link at the bottom of the section explaining everything in depth.